libtenzir/builtins/functions/misc.cpp

```diff
@@ -228,6 +230,55 @@ class length final : public method_plugin {
   }
 };
 
+class has final : public method_plugin {
+public:
+  auto name() const -> std::string override {
+    return "has";
+  }
+
+  auto make_function(invocation inv, session ctx) const
+    -> failure_or<function_ptr> override {
+    auto expr = ast::expression{};
+    auto needle = located<std::string>{};
+    TRY(argument_parser2::method(name())
+          .add(expr, "<expr>")
+          .add(needle, "<string>")
+          .parse(inv, ctx));
+    return function_use::make([needle = std::move(needle),
+                               expr = std::move(expr)](evaluator eval,
+                                                       session ctx) -> series {
+      TENZIR_UNUSED(ctx);
+      auto value = eval(expr);
+      auto f = detail::overload {
+        [&](const arrow::NullArray& array) -> series {
+          auto b = bool_type::make_arrow_builder(arrow::default_memory_pool());
+          check(b->AppendValues(array.length(), false));
+          return {bool_type{}, finish(*b)};
+        },
+          [&](const arrow::StructArray& array) -> series {
+            const auto names = array.struct_type()->fields()
+                               | std::views::transform(&arrow::Field::name);
+            auto b
+              = bool_type::make_arrow_builder(arrow::default_memory_pool());
+            for (auto i = int64_t{0}; i < array.length(); i++) {
+              if (array.IsNull(i)) {
+                check(b->AppendNull());
+                continue;
+              }
+              check(b->Append(std::ranges::find(names, needle.inner) != std::end(names)));
```

Comment:
You can compute the result once outside of the array.

Reply:
True. Done!